src/signal.rs

```diff
@@ -1706,34 +1707,114 @@ mod editable {
     }
 }
 
+pub trait SpinButtonSignals {
+    fn connect_change_value<F>(&self, change_value_func: F) -> u64
+        where F: Fn(&SpinButton, ScrollType) + 'static;
+    fn connect_input<F>(&self, input_func: F) -> u64
+        where F: Fn(&SpinButton) -> Option<Result<f64, ()>> + 'static;
+    fn connect_output<F>(&self, output_func: F) -> u64
+        where F: Fn(&SpinButton) -> bool + 'static;
+    fn connect_value_changed<F>(&self, value_changed_func: F) -> u64
+        where F: Fn(&SpinButton) + 'static;
+    fn connect_wrapped<F>(&self, wrapped_func: F) -> u64
+        where F: Fn(&SpinButton) + 'static;
+}
+
 mod spin_button {
     use SpinButton;
+    use ScrollType;
     use ffi::{GTK_INPUT_ERROR, GtkSpinButton};
     use glib::signal::connect;
     use glib::translate::*;
-    use glib_ffi::{gpointer, GTRUE, GFALSE};
+    use glib_ffi::{GTRUE, GFALSE};
     use libc::{c_int, c_double};
     use std::boxed::Box as Box_;
     use std::mem::transmute;
+    use glib_ffi::gboolean;
+
+    impl ::SpinButtonSignals for SpinButton {
+        fn connect_change_value<F>(&self, change_value_func: F) -> u64
+        where F: Fn(&SpinButton, ScrollType) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&SpinButton, ScrollType) + 'static>> =
+                    Box::new(Box::new(change_value_func));
+                connect(self.to_glib_none().0, "change_value",
+                        transmute(change_trampoline as usize), Box::into_raw(f) as *mut _)
+            }
+        }
 
-    impl SpinButton {
-        pub fn connect_input<F: Fn(&SpinButton, &mut f64) -> Result<bool, ()> + 'static>(&self, f: F) -> u64 {
+        fn connect_input<F>(&self, f: F) -> u64
+        where F: Fn(&SpinButton) -> Option<Result<f64, ()>> + 'static {
             unsafe {
-                let f: Box_<Box_<Fn(&SpinButton, &mut f64) -> Result<bool, ()> + 'static>> = Box_::new(Box_::new(f));
+                let f: Box_<Box_<Fn(&SpinButton) -> Option<Result<f64, ()>> + 'static>> = Box_::new(Box_::new(f));
                 connect(self.to_glib_none().0, "input",
                         transmute(input_trampoline as usize), Box_::into_raw(f) as *mut _)
             }
         }
+
+        fn connect_output<F>(&self, output_func: F) -> u64
+        where F: Fn(&SpinButton) -> bool + 'static {
```

Comment:
> Due `_gtk_boolean_handled_accumulator` in code?

Yes, the documentation fails to make the semantics clear sometimes but the use of this accumulator strongly suggests `Inhibit`.